### PR TITLE
Create a Quest project 

### PR DESCRIPTION
**Pull request by ecotalisman**

### CODE REVIEW

File: src/main/java/service/FirstService.java [hunk 1]
```diff
@@ -0,0 +1,22 @@
+package service;
+
+import models.AnswerChallenge;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import static config.AppConfig.*;
+
+public class FirstService {
```
Comment (Volodymyr95) [inline src/main/java/service/FirstService.java]: not informative name

**Review by Volodymyr95 — APPROVED** [review]

File: src/main/java/config/AppConfig.java [hunk 2]
```diff
@@ -0,0 +1,11 @@
+package config;
+
+public final class AppConfig {
```
Comment (Volodymyr95) [inline src/main/java/config/AppConfig.java]: Constants

File: src/main/java/web/GameChallengeServlet.java [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+public class GameChallengeServlet extends HttpServlet {
+    private static final Logger LOGGER = LoggerFactory.getLogger(GameChallengeServlet.class);
+    GameChallengeService gameChallengeService = new GameChallengeService();
+
+    @Override
+    protected void doPost(HttpServletRequest req, HttpServletResponse resp) throws ServletException, IOException {
+        LOGGER.info("Method doPost() GameChallengeServlet started");
+        AnswerChallenge answer = gameChallengeService.call(Boolean.parseBoolean(req.getParameter("answer")));
+
+        resp.setStatus(201);
```
Comment (Volodymyr95) [inline src/main/java/web/GameChallengeServlet.java]: 201 - when u create something, here return 200

File: src/main/java/web/GameFinishServlet.java [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+public class GameFinishServlet extends HttpServlet {
+    private static final Logger LOGGER = LoggerFactory.getLogger(GameFinishServlet.class);
+    GameFinishService gameFinishService = new GameFinishService();
+
+    @Override
+    protected void doPost(HttpServletRequest req, HttpServletResponse resp) throws ServletException, IOException {
+        LOGGER.info("Method doPost() GameFinishServlet started");
+        AnswerChallenge answer = gameFinishService.call(Boolean.parseBoolean(req.getParameter("answer")));
+
+        resp.setStatus(201);
```
Comment (Volodymyr95) [inline src/main/java/web/GameFinishServlet.java]: 200

File: src/main/java/web/GameStartServlet.java [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+public class GameStartServlet extends HttpServlet {
+    private static final Logger LOGGER = LoggerFactory.getLogger(GameStartServlet.class);
+    GameInitializationService gameInitializationService = new GameInitializationService();
+
+    @Override
+    protected void doPost(HttpServletRequest req, HttpServletResponse resp) throws ServletException, IOException {
+        LOGGER.info("Method doPost() GameStartServlet started");
+        AnswerChallenge answer = gameInitializationService.call(Boolean.parseBoolean(req.getParameter("answer")));
+
+        resp.setStatus(201);
```
Comment (Volodymyr95) [inline src/main/java/web/GameStartServlet.java]: 200

File: src/main/java/service/FirstService.java (hunk 1 above)
Comment (ecotalisman) [reply src/main/java/service/FirstService.java]: It was `Outdated` names, cause three classes have a new name:
- GameInitializationService.java
- GameChallengeService.java
- GameFinishService.java

File: src/main/java/config/AppConfig.java (hunk 2 above)
Comment (ecotalisman) [reply src/main/java/config/AppConfig.java]: changed name to `Constants`

File: src/main/java/web/GameChallengeServlet.java (hunk 3 above)
Comment (ecotalisman) [reply src/main/java/web/GameChallengeServlet.java]: did it

File: src/main/java/web/GameFinishServlet.java (hunk 4 above)
Comment (ecotalisman) [reply src/main/java/web/GameFinishServlet.java]: did it

File: src/main/java/web/GameStartServlet.java (hunk 5 above)
Comment (ecotalisman) [reply src/main/java/web/GameStartServlet.java]: did it

File: src/main/java/constants/Constants.java [hunk 6]
```diff
@@ -0,0 +1,15 @@
+package constants;
+
+public final class Constants {
+    public static final String STEP_START_ACCEPTED = "Ви зайшли в поселення";
+    public static final String STEP_START_NOT_ACCEPTED = "Ви відмовилися заходити в поселення";
+    public static final String STEP_1_ACCEPTED = "1-е завдання було успішно виконане!";
+    public static final String STEP_1_NOT_ACCEPTED = "Сумно, що ви не спробували виконати 1-е завдання";
+    public static final String STEP_2_ACCEPTED = "Вітаю всі завдання були завершені!";
+    public static final String STEP_2_NOT_ACCEPTED = "Отримавши артефакт Око Світу час на пошук Амулету Часу втрачено...";
+    public static final String GAME_START = "jsp/gameStart.jsp";
```
Comment (Volodymyr95) [inline src/main/java/constants/Constants.java]: add for jsp files ending _PAGE or _JSP it makes u code more readable

File: src/main/java/models/AnswerChallenge.java [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+    public AnswerChallenge() {
+    }
+
+    public AnswerChallenge(String message, String page) {
+        this.message = message;
+        this.page = page;
+    }
+
+    public String getMessage() {
+        LOGGER.info("Start method getMessage()");
```
Comment (Volodymyr95) [inline src/main/java/models/AnswerChallenge.java]: not needed logs for getters and setters

File: src/main/java/models/AnswerChallenge.java [hunk 8]
```diff
@@ -0,0 +1,38 @@
+package models;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class AnswerChallenge {
```
Comment (Volodymyr95) [inline src/main/java/models/AnswerChallenge.java]: just Answer

File: src/main/java/service/GameChallengeService.java [hunk 9]
```diff
@@ -0,0 +1,22 @@
+package service;
+
+import models.AnswerChallenge;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import static constants.Constants.*;
+
+public class GameChallengeService {
```
Comment (Volodymyr95) [inline src/main/java/service/GameChallengeService.java]: GameService

Comment (ecotalisman) [reply src/main/java/service/GameChallengeService.java]: did it

File: src/main/java/models/AnswerChallenge.java (hunk 8 above)
Comment (ecotalisman) [reply src/main/java/models/AnswerChallenge.java]: did it

File: src/main/java/models/AnswerChallenge.java (hunk 7 above)
Comment (ecotalisman) [reply src/main/java/models/AnswerChallenge.java]: deleted

File: src/main/java/constants/Constants.java (hunk 6 above)
Comment (ecotalisman) [reply src/main/java/constants/Constants.java]: changed

**Review by Volodymyr95 — APPROVED** [review]

**Pull request merged**